### DISCUSSION TITLE
Best way to attach go terms to blast output

### DISCUSSION THREAD

**Discussion by ocattau** [RobertsLab/resources#1350] [Q&A] [answered]
So I have my blast output from the larval geoduck transcriptome [here](https://gannet.fish.washington.edu/gigas/Panopea-generosa-uniprot_blastx.tab) and I the end goal is to characterize Functional Annotations of the genes [see issue post](https://github.com/RobertsLab/resources/issues/1241).  The format is in 

- target _id
- uniprot identifiers (accession_Id)
- counts (length, eff_length, est_counts, tpm...)

Eventually I want go terms, protein names etc... in order to say something meaningful about geoduck larval development. Is there a tutorial already that shows how to do this?

**Comment by kubu4** [comment, marked as answer]
Unfortunately, I don't think any tutorial/example has been fully fleshed out. There are a few things that have been started, but are incomplete. With that being said, here's my recommendations.

For the most reproducible process, I'd recommend looking at this notebook entry (https://nbviewer.org/github/RobertsLab/code/blob/master/notebooks/sam/20210601_ssal_gff-annotations.ipynb), starting at `In[12]`. Basically, here's the process that needs to take place:

1. Get a newline-delimited file of SP IDs from your BLAST output.
2. Modify the Perl script that's show in `In[13]` to work with SPIDs (replace `P_ENTREZGENEID` with `SWISSPROT`)
3. Run the updated Perl script.
4. Parse out all of the stuff you want (example is shown in `In[18]`).
5. Join this with your BLAST output. There's an example of this in `In[22]`, but you'll need to modify it to work with your BLAST output file.

That will get you a lot of the info you want.

After that, you'll likely want to get GOslims. The next step towards that would be to "unfold" the file you created in Step 5 above. There's a script that will do this for you here: 

- https://github.com/RobertsLab/code/blob/master/script-box/gigas_go_unfold.sh

Once you get through this stuff, we'll tackle getting GOslims!

EDITED: Add forgotten notebook link!

**Comment by sr320** [comment]
It is also ok to "just join" with Swiss-prot annotation table - available via https://d.pr/i/0ZO1bc

see also https://github.com/RobertsLab/code/blob/master/10-blast-2-slim.ipynb

**Reply by ocattau** [reply]
oh okay, thats what I did before but now I see the Go terms. Thanks!
